woof-code/rootfs-skeleton/usr/bin/service2initd

```diff
@@ -0,0 +1,125 @@
+#!/bin/bash
+#Converts systemd .service file into init.d script
+#Written by mistfire
+
+SRVFILE="$1"
+
+if [ "$SRVFILE" == "" ] || [ ! -e "$SRVFILE" ]; then
+ exit
+fi
+
+SRVNAME="$(basename $SRVFILE .service)"
+
+if [ "$(cat $SRVFILE | grep -E '^Type\ =|^Type=' | grep -E 'dbus|udev')"  == "" ]; then
+	 
+	 SRVTYPE="$(cat $SRVFILE | grep '^Type=' | cut -f 2 -d '=' | xargs)"
+	 
+	 EXECCMD="$(cat $SRVFILE | grep '^ExecStart=' | sed -e 's#^ExecStart=##' | xargs)"
+	 PREEXEC="$(cat $SRVFILE | grep '^ExecStartPre=' | sed -e 's#^ExecStartPre=##' | xargs)"
+	 POSTEXEC="$(cat $SRVFILE | grep '^ExecStartPost=' | sed -e 's#^ExecStartPost=##' | xargs)"
+	 EXECSTOP="$(cat $SRVFILE | grep '^ExecStop=' | sed -e 's#^ExecStop=##' | xargs)"
+	 POSTEXECSTOP="$(cat $SRVFILE | grep '^ExecStopPost=' | sed -e 's#^ExecStopPost=##' | xargs)"
+	 RELOADEXEC="$(cat $SRVFILE | grep '^ExecReload=' | sed -e 's#^ExecReload=##' | xargs)"
+
+	 PIDFILE="$(cat $SRVFILE | grep '^PIDFile=' | cut -f 2 -d '=' | xargs)"
+	 KILLSIG="$(cat $SRVFILE | grep '^KillSignal=' | cut -f 2 -d '=' | xargs)"
+	 WKDIR="$(cat $SRVFILE | grep '^WorkingDirectory=' | cut -f 2 -d '=' | xargs)"
+	 ENVFILE="$(cat $SRVFILE | grep '^EnvironmentFile=' | cut -f 2 -d '=' | xargs | sed -e "s#^\-##g" -e "s#^\+##g" -e "s#^\@##g" -e "s#^:##g")"
+	 ENV_VALUES="$(cat $SRVFILE | grep -E '^Environment=' | sed -e 's#^Environment=##' | tr '\n' ' ' | xargs)"
+	 
+	 if [ "$PIDFILE" == "" ]; then
+	  PIDFILE="/var/run/${SRVNAME}.pid"
+	 fi
+
+         if [ "$SRVTYPE" != "oneshot" ]; then
+           BCKPID=' & echo $! > '$PIDFILE
+         else
+           BCKPID=''
+         fi
+	 
+echo '#!/bin/sh
+#script generated by service2initd
+
+ARG1="$1"
+
+start_service(){
+ echo
+'
+    if [ "$SRVTYPE" != "oneshot" ]; then	
+     echo ' rm -f '$PIDFILE' 2>/dev/null'
+    fi
+    
+    #if [ "$ENVFILE" != "" ] && [ -e "$ENVFILE" ]; then
+	 echo ' . '$ENVFILE
+    #fi
+	
+    [ "$WKDIR" != "" ] && echo ' cd '$WKDIR
+    [ "$PREEXEC" != "" ] && echo ' '$PREEXEC
+    
+    if [ "$EXECCMD" != "" ]; then
+     if [ "$ENV_VALUES" != "" ]; then
+      echo ' '$ENV_VALUES' '$EXECCMD''$BCKPID
```

Comment:
Are you assuming that EXECCMD will daemonize? In the systemd world, not daemonizing (so systemd can restart you) is the norm